tests/mocha/blocks/lists_test.js

```diff
@@ -106,4 +106,89 @@ suite('Lists', function() {
     ];
     runSerializationTestSuite(testCases);
   });
+
+  /**
+   * Test cases for serialization where JSON hooks should have null
+   * implementation to avoid serializing xml mutations in json.
+   * @param {!Object} serializedJson basic serialized json
+   * @param {!string} xmlMutation xml mutation that should be ignored/not reserialized in round trip
+   * @return {Array<SerializationTestCase>} test cases
+   */
+  function makeTestCasesForBlockNotNeedingExtraState_(serializedJson, xmlMutation) {
+    return [
+      {
+        title: 'JSON not requiring mutations',
+        json: serializedJson,
+        assertBlockStructure: (block) => {
+          chai.assert.equal(block.type, serializedJson.type);
+        },
+      },
+      {
+        title:
+          'JSON with XML extra state',
+        json: {
+          ...serializedJson,
```

Comment:
I was confused about why a build of a clean checkout of `develop` was resulting in changes to `tests/deps.mocha.js`.  It turns out that this use of `...` meant that `lists_test.js` went from being ES6 to ES2018, and `closure-make-deps` tries to update `testsd/deps.mocha.js` accordingly, but that that update was not included in the same commit.

Not worth fixing now, since PR #6205 is about to delete the deps files from the repo, but a useful reminder to us all to keep an eye out for changes to generated files and try make sure they get checked in at the same time as the change which caused them least we confuse our colleagues…